### PR TITLE
DropBox implementation

### PR DESCRIPTION
**Pull request by Lexsus**
Add DropBox implementation

### CODE REVIEW

File: rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/implementation/DropBoxHelper.java [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+        }
+    }
+
+    public DropBoxHelper() {
+        this.accessToken = "";
+        appInfo = new DbxAppInfo(APP_KEY, APP_SECRET);
+
+    }
+
+    //TODO что такое path - файл на облаке или файл на локальном hdd
```
Comment (vasap87) [inline rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/implementation/DropBoxHelper.java]: Т.к. мы грузим файл по частям, то:
клиентский сервис разбивает файл на куски, передаём байтовый поток, после загрузки получаем url куска на облаке.
При скачивании, передаём url на облаке, получаем байтовый поток.

Премещение, копирование файла, имхо, операция чисто наша внутренняя, в свойствах файла поменять path.

File: rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/implementation/DropBoxImpl.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+import com.google.gson.JsonObject;
+import com.stnetix.ariaddna.externalcloudapi.cloudinterface.IAbstractCloud;
+
+import java.io.File;
+import java.net.URL;
+
+/**
+ * Created by LugovoyAV on 13.09.2017.
+ */
+public class DropBoxImpl implements IAbstractCloud {
```
Comment (kosmoflyko) [inline rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/implementation/DropBoxImpl.java]: What do it this class? This class is contain only delegate method. I thing that all functionality should be move from helper class to this class.

File: rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/implementation/DropBoxHelper.java (hunk 1 above)
Comment (kosmoflyko) [reply rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/implementation/DropBoxHelper.java]: I thing now the team got confused. We have many similar term: FileMetadata, File, VirtualFile, Chunk (files part on cloud)... And we haven't design for modules. Of course it's and my error too ((

File: rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/implementation/DropBoxHelper.java [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+
+/**
+ * Created by LugovoyAV on 06.09.2017.
+ */
+//запрос на аторизацию
+//https://www.dropbox.com/1/oauth2/authorize?response_type=code&client_id=ioiju89iyfxjy4h
+//TODO непонятно пока что возвращать JsonObject
+public class DropBoxHelper {
+    final static String APP_KEY = "ioiju89iyfxjy4h";
+    final static String APP_SECRET = "uys63lipze6ig4v";
```
Comment (kosmoflyko) [inline rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/implementation/DropBoxHelper.java]: Please don't include private information on public resource!

File: rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/implementation/DropBoxHelper.java [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+import java.util.Locale;
+
+/**
+ * Created by LugovoyAV on 06.09.2017.
+ */
+//запрос на аторизацию
+//https://www.dropbox.com/1/oauth2/authorize?response_type=code&client_id=ioiju89iyfxjy4h
+//TODO непонятно пока что возвращать JsonObject
+public class DropBoxHelper {
+    final static String APP_KEY = "ioiju89iyfxjy4h";
```
Comment (kosmoflyko) [inline rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/implementation/DropBoxHelper.java]: Please don't include private information on public resource!

File: rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/implementation/DropBoxHelper.java (hunk 1 above)
Comment (kosmoflyko) [reply rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/implementation/DropBoxHelper.java]: I think we should have managed method for make/delete folder, upload/delete/download files on cloud. But this files it's chunk of real files on client device.

File: rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/implementation/DropBoxHelper.java [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+    }
+
+    //TODO что такое path - файл на облаке или файл на локальном hdd
+    public JsonObject downloadFile(String path)
+    {
+        try
+        {
+            FileOutputStream downloadFile = new FileOutputStream(path);
+            try {
+                FileMetadata metadata = client.files().downloadBuilder("/"+path).download(downloadFile);
```
Comment (kosmoflyko) [inline rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/implementation/DropBoxHelper.java]: This is don't FileMetadata on VUFS, this is chunk. Metadata (in VUFS) it's data in system DB about files properties.
Chunk it is part of some real file on client device. File Splitter Service make chunk and transmit it to Cloud Loader Service. Cloud Loader Service use some API adapter (for example Dropbox) make root directory for ariADDna storage on the cloud. And after it load chunks to the cloud.

File: rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/implementation/DropBoxHelper.java (hunk 4 above)
Comment (Lexsus) [reply rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/implementation/DropBoxHelper.java]: Where we store private information?

File: rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/implementation/DropBoxHelper.java (hunk 3 above)
Comment (Lexsus) [reply rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/implementation/DropBoxHelper.java]: Where we store private information?

File: rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/implementation/DropBoxHelper.java (hunk 4 above)
Comment (kosmoflyko) [reply rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/implementation/DropBoxHelper.java]: We have keystore  for certificates and database for other system information.

File: rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/implementation/DropBoxHelper.java (hunk 1 above)
Comment (Lexsus) [reply rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/implementation/DropBoxHelper.java]: need send to method chunk? ChunkFileEntity or other?

File: rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/implementation/DropBoxImpl.java (hunk 2 above)
Comment (Lexsus) [reply rest-api-cloud-client/src/main/java/com/stnetix/ariaddna/externalcloudapi/implementation/DropBoxImpl.java]: This class will work with the secret information of cloud(oad,save) and with file system, create directories if necessary and other.

**Pull request closed without merging**